mode/clike/clike.js

```diff
@@ -519,6 +519,16 @@ CodeMirror.defineMode("clike", function(config, parserConfig) {
         stream.eatWhile(/[\w\$_]/);
         return "meta";
       },
+      "{": function(stream, state) {
+        // match a single full line with params of multi-line anonymous fn, e.g.:
+        // { a =>\n
+        // { (aa: Int, bb: Int) =>\n
+        var singleLineParams = /^(\w|[:.,() ])+=>$/;
+        if (!stream.match(singleLineParams, true)) // eat the line, if matches
```

Comment:
> when it sees a ">" ahead, it changes the current context to not try and align its body.

how would I change the `current context` ?


also, need to be rather cautious for this not to conflict with other syntax, e.g. if `{` was not present earlier on the same line before `=>`:
```scala
x match {
  case y =>
    khdjhsjdh
}
//or inside fn definition:
def callByReference(argFn: => Unit) = {
  ...
}
```